src/Do.ts

```diff
@@ -170,6 +182,26 @@ class DoClass<M> {
       )
     )
   }
+  let(name: string, a: any): DoClass<M> {
+    return new DoClass(
+      this.M,
+      this.M.map(this.result, s => {
+        const s2 = Object.assign({}, s)
+        s2[name] = a
```

Comment:
The body of this function could be written with fewer operations:
```typescript
s => Object.assign({}, s, { [name]: a })
```

And same for `letL`.